wifiphisher/common/interfaces.py

```diff
@@ -319,233 +339,320 @@ def __init__(self):
         Setup the class with all the given arguments.
 
         :param self: A NetworkManager object
-        :param jamming_argument: The jamming argument given by user
-        :param ap_argument: The AP argument given by user
         :type self: NetworkManager
-        :type jamming_argument: str
-        :type ap_argument: str
         :return: None
         :rtype: None
-        .. seealso:: NetworkAdapter
         """
 
-        # Setup the fields
-        self._interfaces = {}
-        self.ap_iface = ""
-        self.jam_iface = ""
+        self._name_to_object = dict()
+        self._active = set()
 
-        # Create, add and check compatibility for each interface
-        for interface in pyw.interfaces():
-            try:
-                self._interfaces[interface] = NetworkAdapter(interface)
-            except pyric.error as e:
-                pass
+    def is_interface_valid(self, interface_name, mode=None):
+        """
+        Check if interface is valid
+
+        :param self: A NetworkManager object
+        :param interface_name: Name of an interface
+        :param mode: The mode of the interface to be checked
+        :type self: NetworkManager
+        :type interface_name: str
+        :type mode: str
+        :return: True if interface is valid
+        :rtype: bool
+        .. note: The available modes are monitor and AP
+        """
+
+        # raise an error if interface can't be found
+        try:
+            interface_adapter = self._name_to_object[interface_name]
+
+            # raise an error if interface doesn't support the mode
+            if mode == "monitor" and not interface_adapter.has_monitor_mode:
+                raise InvalidInterfaceError(interface_name, mode)
+            elif mode == "AP" and not interface_adapter.has_ap_mode:
+                raise InvalidInterfaceError(interface_name, mode)
 
-    def up_ifaces(self, ifaces):
-        for i in ifaces:
-            card = pyw.getcard(i.get_name())
-            pyw.up(card)
-    
```

Comment:
why `up_ifaces` removed? I remember in `pywifiphisher.py` will call this function, or you have refactored this part too?